lib/Conversion/PolynomialToStandard/PolynomialToStandard.cpp

```diff
@@ -478,21 +481,15 @@ struct ConvertMul : public OpConversionPattern<MulOp> {
         /*bodyBuilder=*/
         [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange args) {
           ImplicitLocOpBuilder b(nestedLoc, nestedBuilder);
-          auto lhs = args[0];
-          auto rhs = args[1];
-          auto accum = args[2];
-          if (!inputOutputHaveSameBitwidth) {
-            lhs = b.create<arith::ExtSIOp>(polymulTensorType.getElementType(),
-                                           lhs);
-            rhs = b.create<arith::ExtSIOp>(polymulTensorType.getElementType(),
-                                           rhs);
-          }
+          auto lhs = b.create<arith::ExtSIOp>(intermediateType, args[0]);
```

Comment:
Question related to these conversions. What are the consequences of doing the conversions inside the linalg generic operation, as opposed to doing this on the entire tensor before and after the generic op? Is there one operation that would allow for better lowering optimizations?

Reply:
I think the effect is rather minimal, since we're mostly using this as a "cheap" way to express the convolution and get a free lowering to loops, rather than expressing any sophisticated transformation of the linalg operation.

But to answer your question, I believe if you put it outside the loop, then it would lower to a sequence of multiple loops, which would then likely be fused together by LLVM optimizations (if we went through llc). But I would have to inspect the lowered code to be sure. (you could try it and use `--mlir-print-ir-after-all` to see the intermediate IRs after each pass.